extensions/cactus-plugin-htlc-coordinator/src/main/typescript/plugin-htlc-coordinator.ts

```diff
@@ -0,0 +1,353 @@
+          const res2 = await pluginHtlc.newContract(newContractRequest);
+          return res2;
+        }
+      }
+      case HtlcPackage.Besu: {
+        const pluginOptions: IPluginHtlcEthBesuOptions = {
+          instanceId: uuidv4(),
+          pluginRegistry,
+        };
+        const factoryHTLC = new PluginFactoryHtlcEthBesu({
+          pluginImportType: PluginImportType.Local,
+        });
+        const pluginHtlc = await factoryHTLC.create(pluginOptions);
+        const request: InitializeRequestBesu = {
+          connectorId: connector.getInstanceId(),
+          keychainId: ownHTLCRequest.keychainId,
+          constructorArgs: ownHTLCRequest.constructorArgs,
+          web3SigningCredential: ownHTLCRequest.web3SigningCredential,
+          gas: ownHTLCRequest.gas,
+        };
+        const res = await pluginHtlc.initialize(request);
+        if (
+          res.transactionReceipt?.status == true &&
+          res.transactionReceipt?.contractAddress != undefined
+        ) {
+          const newContractRequest: NewContractObj = {
+            contractAddress: res.transactionReceipt?.contractAddress,
+            outputAmount: ownHTLCRequest.outputAmount,
+            expiration: ownHTLCRequest.expiration,
+            hashLock: ownHTLCRequest.hashLock,
+            receiver: ownHTLCRequest.receiver,
+            inputAmount: ownHTLCRequest.inputAmount,
+            outputNetwork: ownHTLCRequest.outputNetwork,
+            outputAddress: ownHTLCRequest.outputAddress,
+            connectorId: connector.getInstanceId(),
+            keychainId: ownHTLCRequest.keychainId,
+            web3SigningCredential: ownHTLCRequest.web3SigningCredential,
+            gas: ownHTLCRequest.gas,
+          };
+          const res2 = await pluginHtlc.newContract(newContractRequest);
+          return res2;
+        }
+      }
+      default: {
+        throw new Error(
+          `${fnTag} Unrecognized HTLC Package: ` +
+            `${ownHTLCRequest.htlcPackage} Supported ones are: ` +
+            `${Object.values(HtlcPackage).join(";")}`,
+        );
+      }
+    }
+  }
+
+  public async counterpartyHTLC(
+    counterpartyHTLCRequest: CounterpartyHTLCRequest,
+  ): Promise<InvokeContractV1Response> {
+    const fnTag = `${this.className}#counterpartyHTLC()`;
+    const connector = this.pluginRegistry.plugins.find(
+      (plugin) => plugin.getInstanceId() == counterpartyHTLCRequest.connectorId,
+    ) as PluginLedgerConnectorBesu;
+    const pluginRegistry = new PluginRegistry({});
+    pluginRegistry.add(connector);
+
+    switch (counterpartyHTLCRequest.htlcPackage) {
+      case HtlcPackage.BesuErc20: {
+        const pluginOptions: IPluginHtlcEthBesuErc20Options = {
+          instanceId: uuidv4(),
+          pluginRegistry,
+        };
+
+        const factoryHTLC = new PluginFactoryHtlcEthBesuErc20({
+          pluginImportType: PluginImportType.Local,
+        });
+        const pluginHtlc = await factoryHTLC.create(pluginOptions);
```

Comment:
@jagpreetsinghsasan Please refactor so that the existing plugin registry is used to look up plugin instances instead of creating one from scratch (tests will have to be updated to pass in a plugin registry via the constructor as well - normally the API server takes care of this)